WEB-INTEGRATION.md

```diff
@@ -0,0 +1,742 @@
+`null`, to allow polyfilling events without a dispatch context; although it
+could also be useful for the default behavior of that flag to depend on the
+particular `Event` subclass.
+
+An alternative approach would be to populate the `dispatchSnapshot` property
+when the event is dispatched, with the context in which it is dispatched. This
+is similar to the behavior of `currentTarget` or `eventPhase`. However, the
+dispatch context would stay the same across an event dispatch, whereas those
+properties will change. Furthermore, in most cases where events are created and
+dispatched from JavaScript, the creation and dispatch happen in the same
+context. So we don’t recommend this approach.
+
+### Design principles for dispatch snapshots
+
+In some cases, a single event might have multiple async dispatch contexts as its
+possible causes, because the incoming data flow for that event might have
+multiple branches that go back to different script executions. This is
+particularly important when the data flow in implementations is expected to be
+different from the spec in relevant ways.
+
+An example of this is HTML media playback events, where if the user clicks play,
+and in short succession JS code calls `videoEl.load()` and then `videoEl.play()`
+in two different contexts, all three data flow branches will merge, resulting in
+a single `load` event being fired.
+
+If some of those sources are browser-originated and some originate from JS, only
+the latter ones should be considered. Beyond that, each such individual event
+would have to be considered. For media playback events, this merge is caused by
+debouncing (i.e. if a load is already in progress, calling a method that would
+start one will reuse the existing one), and so the dispatch context should be
+that for the earliest web API call that resulted in this event. But other cases
+might have other needs, and their specifics need to be considered.
+
+## Runtime script errors
+
+The `error` event on a window or worker global object is fired whenever a script
+execution throws an uncaught exception. Having access to the context in which
+the exception was thrown can help determine which of multiple independent
+streams of execution triggered this error, and therefore how to clean up after
+it. Therefore,
+[`ErrorEvent`](https://html.spec.whatwg.org/multipage/webappapis.html#errorevent)
+will have a `throwSnapshot` property, reflecting the context in which the
+exception was thrown.
+
+
+## Promises / async web APIs
+
+The `unhandledrejection` event on a window or worker global object is fired
+whenever a promise resolves with a rejection, without having a rejection handler
+registered. Exposing the context in which the promise was rejected is useful for
+the same reasons as for runtime script errors. Therefore,
+[`PromiseRejectionEvent`](https://html.spec.whatwg.org/multipage/webappapis.html#promiserejectionevent)
+will have a `rejectionSnapshot` property reflecting this context.
```

Comment:
Nit: `throw` is a verb and `rejection` is a noun. It seems better if they were aligned, probably on being a verb.